snapshot_iterator.go

```diff
@@ -109,16 +110,17 @@ func newSnapshotIterator(
 	lastPosition := config.startPosition.Clone()
 
 	iterator := &snapshotIterator{
-		t:    &tomb.Tomb{},
-		data: make(chan fetchData),
-		acks: csync.WaitGroup{}, config: config,
+		t:            &tomb.Tomb{},
+		data:         make(chan fetchData),
+		acks:         csync.WaitGroup{},
+		config:       config,
 		lastPosition: lastPosition,
 	}
 
-	// We first obtain all transaction locks for every table so that we can then
-	// properly get the master position to start cdc mode.
-	// More documentation about why doing it this way at:
-	// https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-snapshots
+	_, err := config.db.ExecContext(ctx, "LOCK TABLES "+strings.Join(config.tables, " READ, ")+" READ")
```

Comment:
There is functional difference between `FLUSH TABLES x, y, z WITH READ LOCK` and I'd suggest you use this vs a simple lock.

See https://dev.mysql.com/doc/refman/8.4/en/flush.html#flush-tables-with-read-lock-with-list